src/sources/nats.rs

```diff
@@ -97,12 +97,28 @@ pub struct NatsSourceConfig {
     /// The `NATS` subject key.
     #[serde(default = "default_subject_key_field")]
     subject_key_field: OptionalValuePath,
+
+    /// The buffer capacity of the underlying NATS subscriber.
+    ///
+    /// This value determines how many messages the NATS subscriber buffers
+    /// before incoming messages are dropped.
+    ///
+    /// See the [async_nats documentation][async_nats_subscription_capacity] for more information.
+    ///
+    /// [async_nats_subscription_capacity]: https://docs.rs/async-nats/latest/async_nats/struct.ConnectOptions.html#method.subscription_capacity
+    #[serde(default = "default_subscription_capacity")]
+    #[derivative(Default(value = "default_subscription_capacity()"))]
+    subscriber_capacity: usize,
 }
 
 fn default_subject_key_field() -> OptionalValuePath {
     OptionalValuePath::from(owned_value_path!("subject"))
 }
 
+const fn default_subscription_capacity() -> usize {
+    4096
```

Comment:
According to [the docs](https://docs.rs/async-nats/latest/async_nats/struct.ConnectOptions.html#method.subscription_capacity) the default is 1024. Shouldn't we stick to that for our default too?

Reply:
From what I can tell, their default is incorrectly documented - [code ref](https://docs.rs/async-nats/latest/src/async_nats/options.rs.html#111).

Reply:
Interesting..